src/snowflake/snowpark/session.py

```diff
@@ -1274,14 +1276,24 @@ def create_dataframe(
             sf_database = self._conn._get_current_parameter("database", quoted=False)
             sf_schema = self._conn._get_current_parameter("schema", quoted=False)
 
+            if schema:
+                if not isinstance(schema, StructType):
+                    raise ValueError(
+                        f"Expects StructType for schema if specified when creating snowpark dataframe from pandas dataframe. Got {type(schema)}"
+                    )
+
+                self._run_query(
```

Comment:
If write_pandas fails, then this table is left without being cleaned? Should we use try/catch to remove this table if write_pandas failed?

Reply:
Nice catch. Let me do that.